### ISSUE TITLE
Error when graphlPort is omitted, but origin.http set.

### ISSUE THREAD

**Issue by digigarlab** [apollographql/apollo-engine-js#138]
Hello,

I am using apollo-engine as a proxy between a React app and a graphql-server.
Since the graphql server is located on a different machine, I am setting the `origin.http` property with the full URL in my config: 

```
{
  engineConfig: ...,
  graphqlPort: 9999, // placeholder, this port number is ignored
  origin: {
    http: {
      url: 'http://some-ip:80/graphql',
    }
  },
  frontend: {
    host: '0.0.0.0',
    port: 8000,
  }
}
```

However I had to give a value to `graphqlPort`, even though [the value is overridden by `origin.http`](https://github.com/apollographql/apollo-engine-js/blob/17a7dc26e0fd3a2c8c64c845f77996230f07c4bb/src/index.ts#L240), otherwise [the engine throws en exception](https://github.com/apollographql/apollo-engine-js/blob/17a7dc26e0fd3a2c8c64c845f77996230f07c4bb/src/index.ts#L176). Am I doing something wrong, or is this a bug?

I am also surprised that `graphqlPort` is attempted to be default to `process.env.PORT`, as it seems more appropriate to use it for `frontend.port`.

**Comment by zionts** [comment]
Hi @digigarlab, thanks for reaching out about this. The default configuration, as you noticed, uses the GraphQL port and the environment PORT variable (if graphqlPort is left unspecified) to inject an origin into the engine configuration and then spawns a random front-end. It then proxies requests that come in to the app twice through the proxy layer – the first trip is simply delivering the request to engine's spawned frontend so that it can pass it to the app and the second trip is the actual parsing and interpretation of the response. In the current release, there is no way to avoid setting default frontends and origins, but in the next release, this problem will be resolved by https://github.com/apollographql/apollo-engine-js/pull/137.

tl;dr: resolved in next release
